upgrade/helpers/tasks.py

```diff
@@ -217,32 +246,57 @@ def _sync_rh_repos_to_satellite(org):
     scl_reposet = entities.RepositorySet(
         name=rhelcontents['rhscl']['repo'].format(os_ver=rhelver), product=scl_product
     ).search()[0]
+    logger.info("Red Hat Software Collection for repos {} is already enabled and found".
+                format(rhelcontents['rhscl']['repo'].format(os_ver=rhelver)))
     try:
         scl_reposet.enable(data={'basearch': arch, 'releasever': '7Server'})
+        logger.info("Red Hat Software collection repository enabled successfully")
     except requests.exceptions.HTTPError:
         logger.warn('Check if repository is already enabled, else retrigger')
     time.sleep(20)
     # Sync enabled Repo from cdn
     scl_repo = entities.Repository(
         name=rhelcontents['rhscl']['repofull'].format(os_ver=rhelver, arch=arch)
     ).search(query={'organization_id': org.id, 'per_page': 100})[0]
-    call_entity_method_with_timeout(entities.Repository(id=scl_repo.id).sync, timeout=2500)
+    attempt = 0
+    while attempt <= 3:
+        try:
+            call_entity_method_with_timeout(
+                entities.Repository(id=scl_repo.id).sync, timeout=2500)
+            break
+        except requests.exceptions.HTTPError as exp:
+            logger.warn("Retry{} after exception: {}".format(attempt, exp))
+            # Wait 10 seconds to reattempt the same retry option
+            time.sleep(10)
+            attempt += 1
     # Enable RHEL 7 Server repository
     server_product = entities.Product(
         name=rhelcontents['server']['prod'], organization=org).search(query={'per_page': 100})[0]
+    logger.info("Product {} is already enabled and found".
+                format(rhelcontents['server']['prod']))
     server_reposet = entities.RepositorySet(
         name=rhelcontents['server']['repo'].format(os_ver=rhelver), product=server_product
     ).search()[0]
+    logger.info("Repository {} is already enabled and found".
+                format(rhelcontents['server']['repo']))
     try:
         server_reposet.enable(data={'basearch': arch, 'releasever': '7Server'})
+        logger.info(
+            "Repository enabled successfully for base arch {}, 7Server".format(arch))
     except requests.exceptions.HTTPError:
         logger.warn('Check if repository is already enabled, else retrigger')
     time.sleep(20)
     # Sync enabled Repo from cdn
     server_repo = entities.Repository(
         name=rhelcontents['server']['repofull'].format(os_ver=rhelver, arch=arch)
     ).search(query={'organization_id': org.id, 'per_page': 100})[0]
-    call_entity_method_with_timeout(entities.Repository(id=server_repo.id).sync, timeout=3600)
+    logger.info("Entities repository sync operation has started successfully"
+                " for name {}".format(server_repo.name))
+    start_time = job_execution_time("Repository sync")
+    call_entity_method_with_timeout(entities.Repository(id=server_repo.id).sync, timeout=6000)
```

Comment:
same here ^^

Reply:
To avoid further any timeout-related failure, we analyzed the log after sync.